price-feeder/oracle/client/chain_height.go

```diff
@@ -0,0 +1,79 @@
+package client
+
+import (
+	"sync"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	rpcclient "github.com/cosmos/cosmos-sdk/client/rpc"
+)
+
+const (
+	// chainHeightCacheInterval represents the amount of seconds
+	// to update the chain height.
+	chainHeightCacheInterval = time.Second * 5
```

Comment:
I think this is too optimistic. There are various edge cases where this won't work, eg, consider the following block height updates (in seconds from 0): 0, 6, 12, 17, ...
the cache will update at 0, 5, 10, 15, 20....

Let's imagine price-feeder will send tx every second. At second 6,7,8, 9, 12, 13,14,15, 17,18,19 the cache will have a wrong value.

Reply:
Is there any way to like subscribe to the node to receive a msg every time a new block is generated?

Reply:
I don't believe so, that's why we need to build a websocket api for osmosis

Reply:
Other question would be, what it could go wrong having the block Height at (real block height - 1) ? according to the tick function that already happens every 1 sec 
@robert-zaremba @adamewozniak

Reply:
Can we use this to subscribe to the new block event?

https://docs.cosmos.network/master/core/events.html#subscribing-to-events

Reply:
Good catch! I will try @brianosaurus thanks

Reply:
I didn't check how the height is used, so don't know what can go wrong if we use a wrong height. I think a tx my be rejected. @adamewozniak can you confirm?

Reply:
Updated to subscribe to event `EventNewBlockHeader` 
@robert-zaremba could you review again?

Reply:
Having the right block height at the right time is very important - could lead to missed votes / bad devx when debugging. Low latency & good voting should be prioritized over any performance.

Reply:
Height is not used as a part of the tx, it's used by the price feeder to time when to submit which txs.